agent/proxycfg/mesh_gateway.go

```diff
@@ -330,6 +420,24 @@ func (s *handlerMeshGateway) handleUpdate(ctx context.Context, u UpdateEvent, sn
 				)
 			}
 
+		case strings.HasPrefix(u.CorrelationID, "discovery-chain:"):
+			resp, ok := u.Result.(*structs.DiscoveryChainResponse)
+			if !ok {
+				return fmt.Errorf("invalid type for response: %T", u.Result)
+			}
+			svcString := strings.TrimPrefix(u.CorrelationID, "discovery-chain:")
+			svc := structs.ServiceNameFromString(svcString)
+
+			if !snap.MeshGateway.IsServiceExported(svc) {
+				delete(snap.MeshGateway.DiscoveryChain, svc)
+				s.logger.Trace("discovery-chain watch fired for unknown service", "service", svc)
+				return nil
+			}
+
+			snap.MeshGateway.DiscoveryChain[svc] = resp.Chain
+
+			// TODO(peering): we need to do this if we are going to setup a cross-partition or cross-datacenter target
```

Comment:
A discovery chain on the exporting side itself may define a cross-datacenter hop somewhere in the chain, so we have to figure out if we're going to let that happen or specifically not let that happen.

The same goes with a cross-partition hop in the chain.

Reply:
Oh hmm good call-out. 

The peering to the middle cluster ends up being functionally transitive, since the mesh gateway accepting the traffic would forward to a different cluster rather than a local service